packages/substrate-client/tests/fixtures.ts

```diff
@@ -0,0 +1,124 @@
+import type { GetProvider } from "@unstoppablejs/provider"
+import { FollowResponse, RpcError, createClient } from "@/."
+import { vi } from "vitest"
+
+export const createTestClient = () => {
+  let onMessage: (msg: string) => void
+  const sendMessage = (msg: {}) => {
+    onMessage(JSON.stringify({ ...msg, jsonrpc: "2.0" }))
+  }
+
+  const receivedMessages: Array<string> = []
+
+  const provider: GetProvider = (_onMessage, _onStatus) => {
+    onMessage = _onMessage
+    return {
+      send(msg) {
+        receivedMessages.push(msg)
+      },
+      open() {
+        _onStatus(0)
+      },
+      close() {},
+    }
+  }
+
+  const client = createClient(provider)
+
+  let latestIdx = 0
+  const getNewMessages = () => {
+    const result = receivedMessages.slice(latestIdx).map((m) => JSON.parse(m))
+    latestIdx = receivedMessages.length
+    return result
+  }
+
+  const getAllMessages = () =>
+    receivedMessages.slice(0).map((m) => JSON.parse(m))
```

Comment:
```suggestion
 const getAllMessages = () => receivedMessages.map((m) => JSON.parse(m))
```

Reply:
good catch! Yep, the `slice(0)` was a leftover b/c in the initial iteration I wasn't applying the `JSON.parse`